src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs

```diff
@@ -8624,5 +8624,43 @@ static void Main()
             await TestAsync(source,
                 MainDescription($"({FeaturesResources.local_variable}) scoped ref int r"));
         }
+
+        [Fact, WorkItem(66854, "https://github.com/dotnet/roslyn/issues/66854")]
+        public async Task TestNullableRefTypeVar1()
+        {
+            var source ="""
+                #nullable enable
+
+                class C
+                {
+                    void M()
+                    {
+                        object? o = null;
+                        $$var s = (string?)o;
+                    }
+                }
+                """;
+            await TestAsync(source,
+                MainDescription($"class System.String?"));
```

Comment:
here we show `class T?`

Reply:
This seems strange to me. Note that we _don't_ do the same thing for explicitly-typed variables, even in annotated code.

Reply:
in an explicitly typed variable, you can see the `?`, so there's no need to.  This is about what to show for 'var'.

Reply:
I'm reaching two conclusions here:

1. Generally, this is good to show `?` to the user.
2. We should simplify the name in Quick Info if the user hovered over `var`, instead of showing the fully-qualified metadata name like you get when you hover over a type name in source.

In other words, I think would expect to see `string?` here, the same as you get hovering over the variable name.